### PR TITLE
[cudapoa] Member type and size unification

### PR DESCRIPTION
**Pull request by atadkase**
- Changed allocation in allocate_block.hpp to infer size and type from data type. 

- Also made similar changes in cudapoa_batch.hpp. 

- Renamed typo concensus to consensus.

### CODE REVIEW

**Comment by atadkase** [comment]
Changed allocation in allocate_block.hpp to infer size and type from data type. Also made similar changes in cudapoa_batch.hpp

File: cudapoa/src/allocate_block.hpp [hunk 1]
```diff
@@ -209,66 +209,66 @@ class BatchBlock
...
-        offset_d_ += cudautils::align<int64_t, 8>(sizeof(uint16_t) * max_nodes_per_window_ * CUDAPOA_MAX_NODE_EDGES * max_poas_);
-        graph_details_d->outgoing_edge_weights = reinterpret_cast<uint16_t*>(&block_data_d_[offset_d_]);
-        offset_d_ += cudautils::align<int64_t, 8>(sizeof(uint16_t) * max_nodes_per_window_ * CUDAPOA_MAX_NODE_EDGES * max_poas_);
-        graph_details_d->sorted_poa = reinterpret_cast<SizeT*>(&block_data_d_[offset_d_]);
-        offset_d_ += cudautils::align<int64_t, 8>(sizeof(SizeT) * max_nodes_per_window_ * max_poas_);
-        graph_details_d->sorted_poa_node_map = reinterpret_cast<SizeT*>(&block_data_d_[offset_d_]);
-        offset_d_ += cudautils::align<int64_t, 8>(sizeof(SizeT) * max_nodes_per_window_ * max_poas_);
-        graph_details_d->sorted_poa_local_edge_count = reinterpret_cast<uint16_t*>(&block_data_d_[offset_d_]);
-        offset_d_ += cudautils::align<int64_t, 8>(sizeof(uint16_t) * max_nodes_per_window_ * max_poas_);
+        offset_d_ += cudautils::align<int64_t, 8>(sizeof(*graph_details_h->nodes) * max_nodes_per_window_ * max_poas_);
```
Comment (r-mafi) [inline cudapoa/src/allocate_block.hpp:225]: should be changed to `sizeof(*graph_details_d->nodes)`

Comment (atadkase) [reply cudapoa/src/allocate_block.hpp:225]: Good catch! Will fix.

Comment (atadkase) [reply cudapoa/src/allocate_block.hpp]: Updated!

**Comment by atadkase** [comment]
rerun tests

**Comment by atadkase** [comment]
rerun tests

**Comment by atadkase** [comment]
rerun tests

**Comment by atadkase** [comment]
rerun tests

**Comment by tijyojwad** [comment]
@atadkase @r-mafi the changelog job is stuck because of a gpuCI change made late last week. Any PRs created _before_ that change will run into the issue. Solution is to merge manually. And since the rest of the tests are passing, I'm doing that.

**Pull request merged**